### DISCUSSION TITLE
Reacting to properties on a foreign object. Is it possible?

### DISCUSSION THREAD

**Discussion by peterje** [pmndrs/jotai#1036] [Q&A]
Hi all,

I have an object that is being updated by code that I cannot modify. I would like to be able to display a property of this object as it updates. So far I have not been able to get it working, is it doable?

Here's a sandbox to demonstrate my issue (simplified):
[Sandbox](https://codesandbox.io/s/jotai-demo-forked-69u25h?file=/src/App.js)

I remember that this was possible in Vue + Vuex, but we are migrating to React + Jotai. I know conceptually that Jotai can't "know" that the object has been updated without re-rendering constantly, but I am hoping there is some solution to this.

```jsx
import React from "react";
import { Provider, atom, useAtom } from "jotai";

const myObject = { number: 0 };
const myObjectAtom = atom(myObject);

const ObjectDisplay = () => {
  const [myObject] = useAtom(myObjectAtom);
  const forceUpdate = React.useState()[1].bind(null, {}); // see NOTE below
  return (
    <div>
      <p>{myObject.number}</p>
      <button onClick={forceUpdate}>Force Update</button>
    </div>
  );
};

const App = () => {
  // I can't change anything from here...
  setInterval(() => {
    myObject.number += 0.1;
  }, 100);
  // ..to here
  return (
    <Provider>
      <ObjectDisplay />
    </Provider>
  );
};

export default App;
```

Thanks,
Peter

**Comment by dai-shi** [comment]
Well, I guess you don't need to make it an atom then. Atom values should be immutable by contract.

You might want to find it interesting how `jotai/valtio` works in a canonical way.
https://codesandbox.io/s/jotai-demo-forked-spp5l4?file=/src/App.js
(I don't think it helps in your scenario though)
